adminguide.rst

```diff
@@ -0,0 +1,514 @@
+job queue and record of past jobs.
+
+
+^^^^^^^
+Systemd
+^^^^^^^
+
+The systemd unit file should be modified to launch the flux broker
+with the option ``--k-ary=N`` where ``N`` is the size of the instance.
+This is required for now to create a flat tree-based overlay network
+for resiliency (all nodes connect back to rank 0). This constraint
+will be removed in the future.
+
+Replicate any modifications on all nodes.
+
+
+.. _instance-administration:
+
+------------------------------
+System Instance Administration
+------------------------------
+
+.. _starting-system-instance:
+
+^^^^^^^^^^^^^
+Starting Flux
+^^^^^^^^^^^^^
+
+The system instance may be started with something like
+
+.. code-block:: console
+
+ $ sudo pdsh -w fluke[1-16] sudo systemctl start flux
+
+Flux brokers can be started in any order or in parallel.
+
+
+^^^^^^^^^^^^^
+Stopping Flux
+^^^^^^^^^^^^^
+
+You can temporarily stop flux by running the following on the rank 0 node:
+
+.. code-block:: console
+
+ $ sudo systemctl stop flux
+
+This kills any running jobs, but preserves the queue of jobs that have
+been submitted but have not yet allocated resources.
+
+The other brokers on other nodes will shut down, then immediately respawn,
+but they will not be online until the rank 0 broker is restarted.
+
+.. _configuration-change:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Changing the Flux configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After changing flux broker or module specific configuration in the TOML
+files under ``/etc/flux``, the configuration may be reloaded with
+
+.. code-block:: console
+
+ $ sudo flux config reload
+
+on each rank where the configuration needs to be updated. The broker will
+reread all configuration files and notify modules that configuration has
+been updated.
+
+Configuration which applies to the ``flux-imp`` or job shell will be reread
+at the time of the next job execution, since these components are executed
+at job launch.
+
+.. _draining-resources:
+
+^^^^^^^^^^^^^^^^^^
+Draining Resources
+^^^^^^^^^^^^^^^^^^
+
+Resources may be temporarily removed from scheduling via the
+``flux resource drain`` command. Currently, resources may only be
+drained at the granularity of rank, for example:
+
+.. code-block:: console
+
+ $ sudo flux resource drain 4
+ $ sudo flux resource list
+      STATE NNODES   NCORES    NGPUS
+       free     15       30        0
+  allocated      0        0        0
+       down      1        2        0
+
+
+To return drained resources use ``flux resource undrain``:
+
+
+.. code-block:: console
+
+ $ sudo flux resource undrain 4
+ $ sudo flux resource list
+      STATE NNODES   NCORES    NGPUS
+       free     16       32        0
+  allocated      0        0        0
+       down      0        0        0
+
+
+.. _queue-admin:
+
+^^^^^^^^^^^^^^^^^^^^^^^
+Managing the Flux Queue
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The queue of jobs is managed by the flux job-manager, which in turn
```

Comment:
Would it be helpful to at least mention the queue listing command `flux jobs`? (maybe reference its man page?)